docs: add documentation about tkms key-gen and crs-gen (#227)
zama-ai/fhevm-backend: docs/getting_started/tkms/create.md

# Request the creation of a new private key

Generating a new set of keys and crs is necessary when creating a new FHE Co-processor.

To do so one can use the TKMS cli tool packaged in the following [docker image](https://github.com/zama-ai/kms-core/pkgs/container/kms-blockchain-simulator).

> The configuration file of the CLI will need to be modified or mounted with a volume to include:

Reviewer: Maybe it is just me because of my lack of ops knowledge; but it is not really clear to me what "mounted with a volume" means here

Author: Just that the config file is accessible inside the docker container.
Not sure how to re-phrase it

Reviewer: Wrote a suggestion

```{toml}
s3_endpoint = ""
object_folder = ["","","",""]

validator_addresses = [""]
http_validator_endpoints = [""]
kv_store_address = ""
faucet_address = ""

asc_address = ""
csc_address = ""
mnemonic = ""
```

## Key generation

### "Insecure"

Insecure key generation is the fastest way to generate a new key.

> The key is generated by a single party and shared with the other parties.

Reviewer: ```suggestion
The key is generated by a single party and shared with the other parties. 
Hence it is not directly insecure, but instead only generated with a centralized trust assumption.
```

```{bash}
cargo run -- -f <path-to-toml-config-file> insecure-key-gen
```

### "Secure"

Secure key generation takes a lot longer and is a two step process.

> For development purposes insecure key generation is the recommended way.

Reviewer: Maybe also just add the command for secure key generation.

Author: Being untested I'm not sure we want to advertise it here do we?

Author: I added the commands in case some user actually wants to try it.

Reviewer: Thanks. I believe we actually want to write these things as extensively as possible; i.e. based on how we want things to be used at the release time. Alternatively we need to reiterate documentation writing for non-breaking changes as well as breaking ones

## Common-Reference-String (CRS) generation

The CRS is a public object used to generate zero-knowledge-proofs of plaintext knowledge (required to add a new ciphertext).

> The `max-num-bits` argument specifies the maximum number of bits provable with a given CRS, usually 2048 is used.

Reviewer: Could you add info here about the secure vs insecure approach, similar to the key generation above?

Author: Ha yes indeed!

Author: Done!

> The `max-num-bits` argument specifies the maximum number of bits provable with a given CRS, usually 2048 is used.

Reviewer: ```suggestion
The `max-num-bits` argument specifies the maximum number of bits provable with a given CRS, usually 2048 is used, since this is the size of the largest data-type currently supported.
```

# Request the creation of a new private key

Generating a new set of keys and crs is necessary when creating a new FHE Co-processor.

To do so one can use the TKMS cli tool packaged in the following [docker image](https://github.com/zama-ai/kms-core/pkgs/container/kms-blockchain-simulator).

> The configuration file of the CLI will need to be modified or mounted with a volume to include:

Reviewer: ```suggestion
The configuration file of the CLI will need to be modified or mounted to a volume accessible within the Docker container. 
The accessible/modified file must include:
```
